### PR TITLE
DOC: Adds tutorials section and move MNIST to notebooks

### PR DESCRIPTION
**Pull request by thomasjpfan**
I think having a tutorials section in the readthedocs that links and briefly describe each example/notebook would be useful to onboard new users.

The notebooks folder would be used for self contained jupyter notebooks, such as MNIST.

The examples folder would be used for more elaborate examples, i.e. projects with one or more scripts.

### CODE REVIEW

File: docs/user/tutorials.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+* `Benchmarks skorch vs pure PyTorch <https://github.com/dnouri/skorch/blob/master/examples/benchmarks/mnist.py>`_ - Compares the performance of skorch and using pure PyTorch on MNIST.
+
+* `Using skorch with Dask <https://github.com/dnouri/skorch/tree/master/examples/rnn_classifer>`_ - Using Dask to parallelize grid search across GPUs.
+
+* `World level language modeling RNN <https://github.com/dnouri/skorch/tree/master/examples/word_language_model>`_ - Uses skorch to train an language model.
+
+* `Seq2Seq Translation using skorch <https://github.com/dnouri/skorch/tree/master/examples/translation>`_ - Translation with a seqeuence to sequence network.
+
+* `Advanced Usage <https://nbviewer.jupyter.org/github/dnouri/skorch/blob/master/notebooks/Basic_Usage.ipynb#Defining-and-training-the-neural-net-classifier>`_ - Dives deep into the inner works of skorch.
```
Comment (taketwo) [inline docs/user/tutorials.rst]: This is the same link as "basic usage"

File: docs/user/tutorials.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+* `Basic Usage <https://nbviewer.jupyter.org/github/dnouri/skorch/blob/master/notebooks/Basic_Usage.ipynb#Defining-and-training-the-neural-net-classifier>`_ - Explores the basics of the skorch api.
+
+* `MNIST with scikit-learn and skorch <https://github.com/dnouri/skorch/blob/master/notebooks/MNIST.ipynb>`_ - Define and train a simple neural network with PyTorch and use it with skorch.
+
+* `Benchmarks skorch vs pure PyTorch <https://github.com/dnouri/skorch/blob/master/examples/benchmarks/mnist.py>`_ - Compares the performance of skorch and using pure PyTorch on MNIST.
+
+* `Using skorch with Dask <https://github.com/dnouri/skorch/tree/master/examples/rnn_classifer>`_ - Using Dask to parallelize grid search across GPUs.
+
+* `World level language modeling RNN <https://github.com/dnouri/skorch/tree/master/examples/word_language_model>`_ - Uses skorch to train an language model.
```
Comment (taketwo) [inline docs/user/tutorials.rst]: train **a** language model

File: docs/user/tutorials.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
+=========
+Tutorials
+=========
+.. _tutorials:
+
+The following are examples and notebooks on how to use skorch.
+
+* `Basic Usage <https://nbviewer.jupyter.org/github/dnouri/skorch/blob/master/notebooks/Basic_Usage.ipynb#Defining-and-training-the-neural-net-classifier>`_ - Explores the basics of the skorch api.
```
Comment (taketwo) [inline docs/user/tutorials.rst]: I think API should be capitalized

**Comment by benjamin-work** [comment]
Thanks @thomasjpfan for the addition and @taketwo for the review. Good work.

**Pull request merged**